### PR TITLE
fix(api): race conditions in eviction

### PR DESCRIPTION
**Pull request by jakubno**
> [!NOTE]
> **Medium Risk**
> Changes core sandbox removal/eviction paths (memory + Redis) and error handling, which could affect lifecycle correctness under concurrency. Risk is mitigated by added eviction-specific checks and expanded tests, but regressions could impact sandbox cleanup and user-visible pause/kill behavior.
> 
> **Overview**
> Tightens sandbox eviction to avoid race conditions by introducing `sandbox.RemoveOpts` (with an `Eviction` flag) and re-checking expiry/transition state under the relevant lock before starting removal; eviction now bails out as `ErrNotEvictable` instead of waiting or removing a sandbox whose TTL was extended. The orchestrator/handlers are updated to use the new removal options and a sentinel `sandbox.ErrNotFound`, and the evictor now suppresses expected eviction failures while logging successful evictions; extensive new tests cover eviction edge cases and retry behavior when `KeepAliveFor` extends `EndTime` mid-transition.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7a929166d0d799ee42764561b05fc7d0c09a73cb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/api/internal/sandbox/errors.go [hunk 1]
```diff
@@ -42,3 +36,5 @@ func (e *NotRunningError) Error() string {
 }
 
 var ErrAlreadyExists = errors.New("sandbox already exists")
+
+var ErrNotEvictable = errors.New("sandbox is not expirable")
```
Comment (arkamar) [inline packages/api/internal/sandbox/errors.go]:
```suggestion
var ErrNotEvictable = errors.New("sandbox is not evictable")
```
This looks like a typo

**Review by jakubno — COMMENTED** [review]
lgtm, thanks for tests @arkamar !

**Review by arkamar — APPROVED** [review]
LGTM
